### PR TITLE
Marketplace

### PR DESCRIPTION
**Pull request by m0g**
Here is the PR for the marketplace module.

The marketplace is a separate view in the liveblog admin that is enabled via environment variable (`MARKETPLACE=true`).

It queries a standalone marketplace apps that returns you a list of given marketers.
The main view is a blog list, with a possibility to filter the items by category or by marketer.
You can then preview a given blog and display the actual embed.

screenshot from 2017-02-13 11-37-34
screenshot from 2017-02-13 11-37-44

### CODE REVIEW

**Comment by mscam** [comment]
:+1:

File: client/app/scripts/liveblog-edit/controllers/blog-settings.js [hunk 1]
```diff
@@ -18,10 +18,15 @@ define([
 ], function(angular, _) {
     'use strict';
     var BlogSettingsController = function($scope, blog, api, blogService, $location, notify,
-        gettext, modal, $q, upload, config, blogSecurityService) {
+        gettext, modal, $q, upload, datetimeHelper, moment, config, blogSecurityService) {
 
         // set view's model
         var vm = this;
+
+        //splitDate: datetimeHelper.splitDateTime(vm.start_date),
```
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/controllers/blog-settings.js]: remove commented code.

File: client/app/index.html [hunk 2]
```diff
@@ -203,7 +203,19 @@ <h3 translate>About</h3>
     <script src="scripts/liveblog-syndication/controllers/producers.js"></script>
     <script src="scripts/liveblog-syndication/directives/producer-edit.js"></script>
     <script src="scripts/liveblog-syndication/directives/producer-list.js"></script>
+
     <script src="scripts/liveblog-freetypes/module.js"></script>
+
+    <!--Liveblog Marketplace-->
+    <script src="scripts/liveblog-marketplace/index.js"></script>
+    <script src="scripts/liveblog-marketplace/controllers/marketplace.js"></script>
+    <!--<script src="scripts/liveblog-marketplace/controllers/marketer.js"></script>-->
```
Comment (nistormihai) [inline client/app/index.html]: remove commented code

File: client/app/scripts/liveblog-edit/controllers/blog-settings.js [hunk 3]
```diff
@@ -161,6 +174,7 @@ define([
                 vm.blogMembers.splice(vm.blogMembers.indexOf(user), 1);
             },
             save: function() {
+                console.log('save', vm.start_date, vm.start_time);
```
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/controllers/blog-settings.js]: console.log isn't needed.

File: client/app/scripts/liveblog-edit/controllers/blog-settings.js [hunk 4]
```diff
@@ -172,6 +186,11 @@ define([
                     original_creator: vm.original_creator._id,
                     blog_status: vm.blog_switch === true? 'open': 'closed',
                     syndication_enabled: vm.syndication_enabled,
+                    market_enabled: vm.market_enabled,
+                    category: vm.category,
+                    //start_date: vm.start_date,
```
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/controllers/blog-settings.js]: remove commented code

File: client/app/scripts/liveblog-marketplace/actions/marketplace.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                getMarketers: function() {
+                    api.get('/marketplace/marketers').then(function(marketers) {
+                        Dispatcher.dispatch({
+                            type: 'ON_GET_MARKETERS',
+                            marketers: marketers
+                        });
+                    });
+                },
+                toggleFilter: function(filters, type, value) {
+                    if (filters.hasOwnProperty(type) && filters[type] == value)
```
Comment (nistormihai) [inline client/app/scripts/liveblog-marketplace/actions/marketplace.js]: use `===`
and I think is better to have just `filters && filters[type] === value`.

File: client/app/scripts/liveblog-marketplace/actions/marketplace.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
+liveblogMarketplace
+    .factory('MarketplaceActions', ['Dispatcher', 'api', '$http', 'lodash',
+        function(Dispatcher, api, $http, _) {
+            return {
+                getBlogs: function(filters) {
+                    api.get('/marketplace/blogs', { where: filters })
```
Comment (nistormihai) [inline client/app/scripts/liveblog-marketplace/actions/marketplace.js]: why not using `apiProvider`?

File: client/app/scripts/liveblog-marketplace/views/marketers-grid.html [hunk 7]
```diff
@@ -0,0 +1,36 @@
+<header>
+    <!--<div sd-searchbar></div>-->
```
Comment (nistormihai) [inline client/app/scripts/liveblog-marketplace/views/marketers-grid.html]: remove code.

File: client/app/scripts/liveblog-marketplace/views/producers-grid.html [hunk 8]
```diff
@@ -0,0 +1,30 @@
+<header>
+    <!--<div sd-searchbar></div>-->
```
Comment (nistormihai) [inline client/app/scripts/liveblog-marketplace/views/producers-grid.html]: remove code.

File: client/app/scripts/liveblog-marketplace/views/producers-grid.html [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+                <div
+                    ng-if="producer.picture_url" 
+                    if-background-image="{{producer.picture_url}}"></div>
+            </figure>
+            <div class="content">
+                <h3>{{ ::producer.name }}</h3>
+                <div class="description">{{producer.email}}</div>
+                <div class="description">{{producer.phone}}</div>
+            </div>
+<!--            <div class="footer">
```
Comment (nistormihai) [inline client/app/scripts/liveblog-marketplace/views/producers-grid.html]: remove code

File: client/app/scripts/liveblog-marketplace/views/search-panel.html [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    <div class="content">
+        <div class="filter-content">
+            <div
+                sd-toggle-box
+                class="filter-title"
+                data-title="{{:: 'Categories' | translate }}"
+                data-open="true">
+                <ul class="filter-box">
+                    <li
+                        ng-repeat="category in categories"
```
Comment (nistormihai) [inline client/app/scripts/liveblog-marketplace/views/search-panel.html]: please use track by category._id

File: client/app/scripts/liveblog-edit/controllers/blog-settings.js [hunk 11]
```diff
@@ -325,16 +344,40 @@ define([
                 });
             }
         }
+
+        var splitDate = moment.tz(vm.newBlog.start_date, config.defaultTimezone);
+        vm.start_date = splitDate.format();
```
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/controllers/blog-settings.js]: why `dateformat` isn't used ?

File: client/app/scripts/liveblog-marketplace/actions/marketplace.js (hunk 6 above)
Comment (m0g) [inline client/app/scripts/liveblog-marketplace/actions/marketplace.js]: If you know how to use `apiProvider` with a nested route, I'll add it.

File: client/app/scripts/liveblog-edit/controllers/blog-settings.js (hunk 11 above)
Comment (m0g) [inline client/app/scripts/liveblog-edit/controllers/blog-settings.js]: Using `dateformat: 'DD/MM/YYYY'` generates an invalid date error

File: client/app/scripts/liveblog-marketplace/views/search-panel.html (hunk 10 above)
Comment (m0g) [inline client/app/scripts/liveblog-marketplace/views/search-panel.html]: Categories is an array of strings, there is no such things as `category._id`

**Pull request merged**